mesonbuild/environment.py

```diff
@@ -708,6 +709,22 @@ def _guess_nix_linker(compiler: typing.List[str], for_machine: MachineChoice,
         v = search_version(o)
         if o.startswith('LLD'):
             linker = LLVMDynamicLinker(compiler, for_machine, 'lld', prefix, version=v)  # type: DynamicLinker
+        elif e.startswith('lld-link: '):
+            # Toolchain wrapper got in the way; this happens with e.g. https://github.com/mstorsjo/llvm-mingw
+            # Let's try to extract the linker invocation command to grab the version.
+
+            _, o, e = Popen_safe(compiler + check_args + ['-v'])
+
+            try:
+                linker_cmd = re.match(r'.*\n(.*?)\nlld-link: ', e, re.DOTALL).group(1)
+                linker_cmd = shlex.split(linker_cmd)[0]
+            except (AttributeError, IndexError, ValueError):
```

Comment:
I think this should be:
```python
m = re.match(r'.*\n(.*?)\nlld-link: ', e, re.DOTALL)
if m:
    linker_cmd = shlex.split(m.group(1))[0]
    _, o, _ = Popen_safe(...)
    v = version_search(o)
```

That should avoid all of the exceptions and remove the odd looking try/except/else.

Reply:
That's not really equivalent. `shlex.split` may throw a `ValueError` if it doesn't like the syntax; its pretty unlikely but I'd rather not crash meson in that situation.

Reply:
I still really dislike the empty `except` block followed by an `else` block. it just seems odd. Let me sit on it a bit.